modules/apps/object/object-service/src/main/java/com/liferay/object/internal/field/business/type/RelationshipObjectFieldBusinessType.java

```diff
@@ -93,63 +95,106 @@ public Object getValue(
 					NAME_OBJECT_RELATIONSHIP_ERC_OBJECT_FIELD_NAME,
 				objectField);
 
+		String objectRelationshipObjectFieldName = StringUtil.split(
+			objectField.getName(), CharPool.UNDERLINE
+		).get(
+			1
+		);
+
 		if (Objects.equals(
 				objectField.getRelationshipType(),
 				ObjectRelationshipConstants.TYPE_ONE_TO_MANY) &&
-			values.containsKey(objectRelationshipERCObjectFieldName)) {
+			values.containsKey(objectRelationshipObjectFieldName)) {
 
-			String externalReferenceCode = GetterUtil.getString(
-				values.get(objectRelationshipERCObjectFieldName));
+			if (!(values.get(objectRelationshipObjectFieldName) instanceof
```

Comment:
I don't get why is necessary adding code specific to the nested field since it is processed later down the road. I took out this part of the code and it still worked as expected.

Reply:
Some other relevant test failed without it but still made a chang in the logic.